### PR TITLE
exec: add support for `Env`

### PR DESCRIPTION
**Pull request by TomasTomecek**
Signed-off-by: Tomas Tomecek <ttomecek@redhat.com>

### CODE REVIEW

**Comment by TomasTomecek** [comment]
Test failure means that the docker engine is not new enough. Shall I skip it?

**Comment by TomasTomecek** [comment]
Janky is stuck :/

**Comment by shin-** [comment]
It's a Jenkins/webhook config issue - we don't actually run the `janky` tests anymore.

File: docker/api/exec_api.py [hunk 1]
```diff
@@ -50,7 +61,8 @@ def exec_create(self, container, cmd, stdout=True, stderr=True,
             'AttachStdin': stdin,
             'AttachStdout': stdout,
             'AttachStderr': stderr,
-            'Cmd': cmd
+            'Cmd': cmd,
+            'Env': environment or [],
```
Comment (shin-) [inline docker/api/exec_api.py]: What purpose does `or []` serve here? `"Env": None` should be fine.

Comment (TomasTomecek) [inline docker/api/exec_api.py]: I was worried how `'Env': None` gets serialized to json, API spec said it should be array.

I trust you 🍸

**Comment by TomasTomecek** [comment]
Thanks @shin-, now I understand it. Should be fixed.

**Comment by shin-** [comment]
Thank you!

**Pull request merged**